test/test_command.py

```diff
@@ -258,39 +258,83 @@ def setUp(self):
 
-        self.command.remove_extra_packages(packages)
+        self.assertEqual(("django-nose",), self.command.determine_extra_packages(packages))
 
-        mock_check_call.assert_called_once_with(['pip', 'uninstall', '-y', 'django-nose'])
+        self.assertFalse(mock_check_call.called)
 
     @patch('subprocess.check_output')
     @patch('subprocess.check_call')
-    def test_remove_extra_packages_with_dashe_directive(self, mock_check_call, mock_check_output):
+    def test_determine_extra_packages_with_dashe_directive(self,
+                                                           mock_check_call, mock_check_output):
         mock_check_output.return_value = \
-            'mock==1.2\nDjango==1.7\nnose==1.3\n ' \
-            '-e http://example.com/some-repo.git\n ' \
+            'mock==1.2\nDjango==1.7\nnose==1.3\n' \
+            '-e http://example.com/some-repo.git\n' \
             'django-nose==1.0\n'
         packages = _create_packages('mock==1.2\nDjango==1.7\nnose==1.3\n')
 
+        self.assertEqual(("django-nose",), self.command.determine_extra_packages(packages))
+
+        self.assertFalse(mock_check_call.called)
+
+
+class TestRemoveExtra(unittest.TestCase):
+
+    def setUp(self):
+        self.parser = cli.create_parser()
+        self.rc_file_blank = tempfile.NamedTemporaryFile()
+        self.command = command.Command(self.parser.parse_args([]), self.rc_file_blank.name)
+
+    @patch('subprocess.check_output')
+    @patch('subprocess.check_call')
+    def test_remove_extra_packages_when_none_to_remove(self, mock_check_call, mock_check_output):
+        mock_check_output.return_value = 'mock==1.2\nDjango==1.7\nnose==1.3\n'
+        packages = _create_packages()
+
+        self.command.remove_extra_packages(packages)
+
+        self.assertFalse(mock_check_call.called)
+
+    @patch('subprocess.check_output')
+    @patch('subprocess.check_call')
+    def test_remove_extra_packages_when_some_to_remove(self, mock_check_call, mock_check_output):
+        mock_check_output.return_value = 'mock==1.2\nDjango==1.7\nnose==1.3\ndjango-nose==1.0\n'
+        packages = _create_packages()
+
         self.command.remove_extra_packages(packages)
 
         mock_check_call.assert_called_once_with(['pip', 'uninstall', '-y', 'django-nose'])
 
     @patch('subprocess.check_output')
     @patch('subprocess.check_call')
+    def test_run_remove_extra_packages_dry_run(self, mock_check_call, mock_check_output):
+        mock_check_output.return_value = 'mock==1.2\nDjango==1.7\nnose==1.3\ndjango-nose==1.0\n'
+        package_file = tempfile.NamedTemporaryFile()
+        package_file.write(b'mock==1.2\nDjango==1.7\nnose==1.3\n')
+        package_file.seek(0)
+        self.command.args = self.parser.parse_args(['-xn', package_file.name])
+
+        self.command.run()
+
+        expected = "The following packages would be removed:\n    django-nose"
+        self.assertEqual(expected, sys.stdout.getvalue().strip())
```

Comment:
In the preexisting `TestRemoveExtra` tests, the subprocess calls are mocked out, but the stdout output isn't.  (Nose and unittest conspire to suppress the output if the test passes, but AFAICT, pipreq isn't doing anything special with it).

Same with this test.  It looks different even so, though, because:
* in this one, `pip uninstall` is never run (and indeed, the test uses `mock_check_call` to verify that)
* the stdout output *is* verified, which it isn't in the other tests.  (Fair enough.  In those, the stdout output is secondary; it's only the `pip uninstall` call -- or absence thereof -- that really matters.  In this one, though, the stdout output is the whole point of this patch, so of course I verify it.)

That's my thinking anyway.  Is there something I've missed?